hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python
+import asyncio
+import logging
+import aiohttp
+import ujson
+import websockets
+import hummingbot.connector.exchange.bitmart.bitmart_constants as CONSTANTS
+
+from typing import Optional, List, Dict, Any, AsyncIterable
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.logger import HummingbotLogger
+from hummingbot.connector.exchange.bitmart import bitmart_utils
+from hummingbot.connector.exchange.bitmart.bitmart_order_book import BitmartOrderBook
+from hummingbot.core.api_throttler.async_throttler import AsyncThrottler
+
+
+class BitmartAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    MESSAGE_TIMEOUT = 10.0
+    SNAPSHOT_TIMEOUT = 60 * 60  # expressed in seconds
+    PING_TIMEOUT = 10.0
+
+    _logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._logger is None:
+            cls._logger = logging.getLogger(__name__)
+        return cls._logger
+
+    def __init__(self, throttler: Optional[AsyncThrottler] = None, trading_pairs: List[str] = None):
+        super().__init__(trading_pairs)
+        self._throttler = throttler or self._get_throttler_instance()
+        self._snapshot_msg: Dict[str, any] = {}
+
+    @classmethod
+    def _get_throttler_instance(cls) -> AsyncThrottler:
+        throttler = AsyncThrottler(CONSTANTS.RATE_LIMITS)
+        return throttler
+
+    @classmethod
+    async def get_last_traded_prices(cls, trading_pairs: List[str]) -> Dict[str, float]:
+        result = {}
+        async with aiohttp.ClientSession() as client:
+            async with client.get(f"{CONSTANTS.REST_URL}/{CONSTANTS.GET_LAST_TRADING_PRICES_PATH_URL}", timeout=10) as response:
+                response_json = await response.json()
+                for ticker in response_json["data"]["tickers"]:
+                    t_pair = bitmart_utils.convert_from_exchange_trading_pair(ticker["symbol"])
+                    if t_pair in trading_pairs and ticker["last_price"]:
+                        result[t_pair] = float(ticker["last_price"])
+        return result
```

Comment:
You are not consuming the `AsyncThrottler` at all here.  Wrap the selection in an async context as you did for `BitmartExchange`

```suggestion
        async with self._throttler.execute_task(CONSTANTS.GET_LAST_TRADING_PRICES_PATH_URL):
            async with aiohttp.ClientSession() as client:
                async with client.get(f"{CONSTANTS.REST_URL}/{CONSTANTS.GET_LAST_TRADING_PRICES_PATH_URL}", timeout=10) as response:
                    response_json = await response.json()
                    for ticker in response_json["data"]["tickers"]:
                        t_pair = bitmart_utils.convert_from_exchange_trading_pair(ticker["symbol"])
                        if t_pair in trading_pairs and ticker["last_price"]:
                            result[t_pair] = float(ticker["last_price"])
        return result
```

Reply:
Hi @CrimsonJacket, I have added the api throttler encapsulations to all the rest api functions in order book data source file.

However, adding api throttler to ws functions in _order_book_data_source or _user_stream_data_source with exchange specified limits, arise warning of api exceeding limits, whereas leaving them as with current timeout parameters and configurations doesn't raise any errors, works seamlessly. So, haven't extended the api throttler to ws.

Please check the made changes and review.

Thanks!